engine/src/assets.rs

```diff
@@ -12,6 +14,10 @@ impl<S: States> Plugin for AssetsPlugin<S> {
         app.add_loading_state(LoadingState::new(
             self.loading_state.clone(),
         ));
+        // add custom asset types
+        app.add_plugin(RonAssetPlugin::<
+            self::animation::SpriteAnimation,
+        >::new(&["anim.ron"]));
```

Comment:
Decide whether we want to use RON format, or TOML, or something else. I have a preference for TOML, but it doesn't play nice with rust enums.

TODO: try out all the syntax in a RON file and see what it looks/feels like. If it is cumbersome to work with, switch to TOML and change the asset format serde structs to something that would work nicely with TOML. Might need an extra processing layer.